### PR TITLE
Docs/update dg

### PR DESCRIPTION
**Pull request by Yongbeom-Kim**
Update dg!

### CODE REVIEW

**Review by Aishwarya-Hariharan-Iyer — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -94,9 +94,10 @@ Here's a (partial) class diagram of the `Logic` component:
 
 How the `Logic` component works:
 1. When `Logic` is called upon to execute a command, it uses the `AddressBookParser` class to parse the user command to result in a `Command` object (more precisely, an object of one of its subclasses e.g., `AddClientCommand`). More specifically,
-   1. If the command is specific to an entity (e.g. `Project`, `Client` or `Issue`), this results in a `CommandParser` object (More specifically, an object of one of its subclasses e.g., `ClientCommandParser`)), which then yields a `Command` object.
+   1. If the command is specific to an entity (e.g. `Project`, `Client` or `Issue`), this results in a `CommandParser` object (More specifically, an object of one of its subclasses e.g., `ClientCommandParser`)).
+      1. This `XYZCommandParser` object then yields a `Command` object (more specifically, a derived class of `ProjectCommand`, `ClientCommand` or `IssueCommand` respectively.
```
Comment (Donovan9617) [inline docs/DeveloperGuide.md:98]: Missing a closing )

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -94,9 +94,10 @@ Here's a (partial) class diagram of the `Logic` component:
 
 How the `Logic` component works:
 1. When `Logic` is called upon to execute a command, it uses the `AddressBookParser` class to parse the user command to result in a `Command` object (more precisely, an object of one of its subclasses e.g., `AddClientCommand`). More specifically,
-   1. If the command is specific to an entity (e.g. `Project`, `Client` or `Issue`), this results in a `CommandParser` object (More specifically, an object of one of its subclasses e.g., `ClientCommandParser`)), which then yields a `Command` object.
+   1. If the command is specific to an entity (e.g. `Project`, `Client` or `Issue`), this results in a `CommandParser` object (More specifically, an object of one of its subclasses e.g., `ClientCommandParser`)).
```
Comment (Donovan9617) [inline docs/DeveloperGuide.md:97]: Extra ) at the end

File: docs/diagrams/ParserPartialInitialisationSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+deactivate ProjectWithoutModel
+
+create AddProjectCommand
+ProjectCommandParser --> AddProjectCommand : AddProjectCommand(projectWithoutModel)
+activate AddProjectCommand
+
+AddProjectCommand --> ProjectCommandParser
+deactivate AddProjectCommand
+
+activate LogicManager
```
Comment (Donovan9617) [inline docs/diagrams/ParserPartialInitialisationSequenceDiagram.puml:32]: Activation bar shouldnt begin before execute is executed based on the image

**Review by Donovan9617 — COMMENTED** [review]
Couple of changes required for the Parser Initialiser Sequence Diagram with other minor fixes

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Yongbeom-Kim) [reply docs/DeveloperGuide.md]: Addressed

File: docs/diagrams/ParserPartialInitialisationSequenceDiagram.puml (hunk 3 above)
Comment (Yongbeom-Kim) [reply docs/diagrams/ParserPartialInitialisationSequenceDiagram.puml:32]: Addressed

**Pull request merged**